frontend/openchat-client/src/openchat.ts

```diff
@@ -3088,69 +3083,73 @@ export class OpenChat extends OpenChatAgentWorker {
         let threadRootMessageIndex: number | undefined = undefined;
-        let message: EventWrapper<Message>;
+        let eventIndex: number;
+        let isReaction = false;
         switch (notification.kind) {
             case "channel_notification": {
                 chatId = notification.chatId;
                 threadRootMessageIndex = notification.threadRootMessageIndex;
-                message = notification.message;
+                eventIndex = notification.eventIndex;
                 break;
             }
             case "direct_notification": {
                 chatId = notification.sender;
-                threadRootMessageIndex = notification.threadRootMessageIndex;
-                message = notification.message;
+                eventIndex = notification.eventIndex;
                 break;
             }
             case "group_notification": {
                 chatId = notification.chatId;
                 threadRootMessageIndex = notification.threadRootMessageIndex;
-                message = notification.message;
+                eventIndex = notification.eventIndex;
                 break;
             }
             case "channel_reaction": {
                 chatId = notification.chatId;
                 threadRootMessageIndex = notification.threadRootMessageIndex;
-                message = notification.message;
+                eventIndex = notification.messageEventIndex;
+                isReaction = true;
                 break;
             }
             case "direct_reaction": {
                 chatId = notification.them;
-                message = notification.message;
+                eventIndex = notification.messageEventIndex;
+                isReaction = true;
                 break;
             }
             case "group_reaction": {
                 chatId = notification.chatId;
                 threadRootMessageIndex = notification.threadRootMessageIndex;
-                message = notification.message;
+                eventIndex = notification.messageEventIndex;
+                isReaction = true;
                 break;
             }
-            case "added_to_group_notification":
             case "added_to_channel_notification":
                 return;
         }
 
-        if (threadRootMessageIndex !== undefined) {
-            // TODO fix this for thread messages
+        const serverChat = this._liveState.serverChatSummaries.get(chatId);
+        if (serverChat === undefined) {
             return;
         }
 
-        const serverChat = this._liveState.serverChatSummaries.get(chatId);
+        const minVisibleEventIndex = serverChat.kind === "direct_chat" ? 0 : serverChat.minVisibleEventIndex;
+        const latestClientEventIndex = Math.max(eventIndex, serverChat.latestEventIndex);
 
-        if (serverChat === undefined || serverChat.latestEventIndex >= message.index) {
-            return;
+        if (isReaction) {
+            // TODO first clear the existing cache entry
```

Comment:
Is this not really important?

Reply:
Its not urgent but I left the TODO just for future reference